src/components/[guild]/RolePlatforms/RolePlatforms.tsx

```diff
@@ -124,6 +139,45 @@ const RolePlatforms = ({ roleId }: Props) => {
                     )
                   }
                   actionRow={PlatformCardSettings && <PlatformCardSettings />}
+                  contentRow={
+                    CAPACITY_TIME_PLATFORMS.includes(type) && (
+                      <Wrap>
+                        {showCapacityTimeTags && (
+                          <CapacityTimeTags rolePlatform={rolePlatform} />
+                        )}
+
+                        <EditRolePlatformCapacityTimeButton
```

Comment:
There're invidual buttons and modals mounted for each reward now, but only a single state, so if there're multiple CAPACITY_TIME supporting rewards in a role, and I click on edit, every modal is going to open. I think we should abstract the modal and the state into EditRewardAvailibilityButton (EditRolePlatformCapacityTimeButton now), or as I'm seeing there's CapacityTimeSetup that abstracts just something like that, why not use that?

Reply:
Uh, awesome idea, thanks, made a refactor in https://github.com/agoraxyz/guild.xyz/pull/946/commits/927de57a884fa0f7fe50e12bc7ddfc5633f58863! 👌 I think it became much cleaner, and I also managed to remove the outer `Wrap`s in a couple of places, let me know if you like it this way!